src/TestFramework/TestFramework/Attributes/DataSource/DataRowAttribute.cs

```diff
@@ -95,15 +95,24 @@ public DataRowAttribute(params object?[]? data)
     /// </summary>
     private static string? GetObjectString(object? obj)
     {
+        if (TestIdGenerationStrategy != TestIdGenerationStrategy.FullyQualified)
+        {
+            return obj?.ToString();
+        }
+
         if (obj == null)
         {
-            return null;
+            return "null";
         }
 
-        if (TestIdGenerationStrategy != TestIdGenerationStrategy.FullyQualified
-            || !obj.GetType().IsArray)
+        if (!obj.GetType().IsArray)
```

Comment:
Not for this PR but it could be good to check `is IEnumerable` instead so we would also show different collections. This being said, this code is only for `DataRow` so I am not sure what kind of collection is allowed to be passed on attributes.